scripts/ci/pre_commit/pre_commit_check_pre_commit_hook_names.py

```diff
@@ -50,16 +57,111 @@ def main() -> int:
     return retval
 
 
-def get_errors(content, max_length):
+def get_errors_and_ids(content, max_length):
     errors = []
+    ids = set()
     for repo in content['repos']:
         for hook in repo['hooks']:
+            if 'id' in hook:
+                id = hook['id']
+                ids.add(id)
             if 'name' not in hook:
                 continue
             name = hook['name']
             if len(name) > max_length:
                 errors.append(name)
-    return errors
+    return errors, list(ids)
+
+
+def render_template(
+    searchpath: Path,
+    template_name: str,
+    context: Dict[str, Any],
+    extension: str,
+    autoescape: bool = True,
+    keep_trailing_newline: bool = False,
+) -> str:
+    """
+    Renders template based on its name. Reads the template from <name>_TEMPLATE.md.jinja2 in current dir.
+    :param template_name: name of the template to use
+    :param context: Jinja2 context
+    :param extension: Target file extension
+    :param autoescape: Whether to autoescape HTML
+    :param keep_trailing_newline: Whether to keep the newline in rendered output
+    :return: rendered template
+    """
+    import jinja2
+
+    template_loader = jinja2.FileSystemLoader(searchpath=searchpath)
+    template_env = jinja2.Environment(
+        loader=template_loader,
+        undefined=jinja2.StrictUndefined,
+        autoescape=autoescape,
+        keep_trailing_newline=keep_trailing_newline,
+    )
+    template = template_env.get_template(f"{template_name}_TEMPLATE{extension}.jinja2")
+    content: str = template.render(context)
+    return content
+
+
+# def get_precommit_jinja_context(pre_commit_ids: List) -> Dict[str, List]:
+#     context: Dict[str, List] = {"PRE_COMMIT_IDS": pre_commit_ids}
+#     return context
+
+
+def get_target_precommit_parent_dir():
+    airflow_source = Path.cwd()
+    pre_commit_ids_file = Path(airflow_source) / "dev" / "breeze" / "src" / "airflow_breeze"
+    return pre_commit_ids_file
+
+
+def get_breeze_pyproject_toml_dir():
+    airflow_source = Path.cwd()
+    breeze_pyproject_toml_file = Path(airflow_source) / "dev" / "breeze"
+    return breeze_pyproject_toml_file
+
+
+@lru_cache(maxsize=None)
+def black_mode():
+    from black import Mode, parse_pyproject_toml, target_version_option_callback
+
+    config = parse_pyproject_toml(Path(get_breeze_pyproject_toml_dir(), "pyproject.toml"))
+
+    target_versions = set(
+        target_version_option_callback(None, None, config.get('target_version', [])),
```

Comment:
I think it should be `config.get('target_version', set())`
The function signature expect Tuple not List
https://github.com/psf/black/blob/d038a24ca200da9dacc1dcb05090c9e5b45b7869/src/black/__init__.py#L154

@potiuk @Bowrna

Reply:
Ah. Yeah. I think it's the result of Black change - it's been released yesterday!

Reply:
https://pypi.org/project/black/22.1.0/

Reply:
PR: https://github.com/apache/airflow/pull/21215